rosbag2_compression/src/rosbag2_compression/sequential_compression_reader.cpp

```diff
@@ -155,12 +155,16 @@ void SequentialCompressionReader::load_next_file()
     throw std::runtime_error{"Cannot load next file; already on last file!"};
   }
 
-  if (decompressor_) {
-    throw std::runtime_error{"Bagfile is not opened"};
+  if (compression_mode_ == rosbag2_compression::CompressionMode::NONE) {
+    throw std::runtime_error{"cannot use SequentialCompressionWriter with NONE compression."};
   }
 
   ++current_file_iterator_;
   if (compression_mode_ == rosbag2_compression::CompressionMode::FILE) {
+    if (decompressor_ == nullptr) {
+      throw std::runtime_error{"Bagfile is not opened."};
```

Comment:
Can we please make this more descriptive? E.g. Attempted to decompress file "f" but the decompressor was not initialized (nullptr)..

Reply:
It can't include the file name since the function doesn't have that information.
I introduced more verbose logs elsewhere in `SequentialCompressionReader`.